From c4ea0e0af304d4ff399839cfb1ec76012f183ccf Mon Sep 17 00:00:00 2001
From: "J. R. Okajima" <hooanon05g@gmail.com>
Date: Thu, 1 Aug 2019 11:17:48 +0900
Subject: [PATCH 1/2] aufs: minor, for userspace, don't use typeof

This is detected by the commit in v5.3-rc1,
	d6fc9fcbaa65 2019-07-08 kbuild: compile-test exported headers to
		ensure they are self-contained
But the issue is not for v5.3-rc1 only I am afraid.  So this commit is
brought to aufs4.14 which is my current development base.

Signed-off-by: J. R. Okajima <hooanon05g@gmail.com>
(cherry picked from commit d38615f7fef1c65e30e3cc418daa9e19c93ed98f)
---
 include/uapi/linux/aufs_type.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/uapi/linux/aufs_type.h b/include/uapi/linux/aufs_type.h
index 1d9bf5e..231f87c 100644
--- a/include/uapi/linux/aufs_type.h
+++ b/include/uapi/linux/aufs_type.h
@@ -241,7 +241,11 @@ enum {
 
 /* borrowed from linux/include/linux/kernel.h */
 #ifndef ALIGN
+#ifdef _GNU_SOURCE
 #define ALIGN(x, a)		__ALIGN_MASK(x, (typeof(x))(a)-1)
+#else
+#define ALIGN(x, a)		(((x) + (a) - 1) & ~((a) - 1))
+#endif
 #define __ALIGN_MASK(x, mask)	(((x)+(mask))&~(mask))
 #endif
 
-- 
2.22.0.545.g9c9b961d7e


From ba5f036b095583e48446c0b11db1faa69be3d948 Mon Sep 17 00:00:00 2001
From: "J. R. Okajima" <hooanon05g@gmail.com>
Date: Sat, 3 Aug 2019 23:20:04 +0900
Subject: [PATCH 2/2] aufs5.2 20190805

Signed-off-by: J. R. Okajima <hooanon05g@gmail.com>
---
 include/uapi/linux/aufs_type.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/uapi/linux/aufs_type.h b/include/uapi/linux/aufs_type.h
index 6ad3e53..6fe7b75 100644
--- a/include/uapi/linux/aufs_type.h
+++ b/include/uapi/linux/aufs_type.h
@@ -40,7 +40,7 @@
 
 #include <linux/limits.h>
 
-#define AUFS_VERSION	"5.2-20190715"
+#define AUFS_VERSION	"5.2-20190805"
 
 /* todo? move this to linux-2.6.19/include/magic.h */
 #define AUFS_SUPER_MAGIC	('a' << 24 | 'u' << 16 | 'f' << 8 | 's')
-- 
2.22.0.545.g9c9b961d7e

